### ISSUE TITLE
SSH times out while waiting for machine to boot

### ISSUE THREAD

**Issue by sarahmonster** [Automattic/vip-quickstart#516]
VIP Quickstart was actually working really smoothly on my machine for a few months, and now it's started acting up again. I killed my old install, thinking if I started fresh it should work again, but I still keep having the same problem, and I'm not sure what's causing it.

Basically it gets stuck while trying to connect to SSH:

`Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.`

If I try to `vagrant ssh` I get this error:
`ssh_exchange_identification: Connection closed by remote host`

Is there something I can do to fix this? I've tried removing everything and setting up a new instance quite a few times, but I just keep having the same problem.

**Comment by joshbetz** [comment]
Are you on the latest version of Quickstart? This sounds like the same issue that we fixed in #502.

**Comment by sarahmonster** [comment]
Yep. I cloned a fresh copy and tried to start all over—but I still get the same error. I've checked my vagrantfile and it has the necessary `config.ssh.insert_key = false` line. 

Previously I was sometimes able to fix the issue (actually notably the issue solved by #502) by setting `v.gui = true` and manually entering a username/password each time vagrant booted up, but that no longer works here—I still get the timeout error.

**Comment by ronrlin** [comment]
I'm having this problem as well on a Mac Mid-2014 running El Capitan.

**Comment by Tarendai** [comment]
@sarahmonster I made some changes that corrected the timeout issue in most cases for myself and some others in https://github.com/Automattic/vip-quickstart/commit/d312b3b4754d2e48fb886ea973e76fd62bb4d923 but I'm concerned that may not resolve the issue for you as you reported `vagrant ssh` isn't working.

Can you confirm if that's still the case with the Vagrant switched from SSH key to SSH user/pass auth?

**Comment by sarahmonster** [comment]
I seem to still be having the same issue—I just tested with a fresh Quickstart install, and I get:

```
default: SSH address: 127.0.0.1:2200
default: SSH username: vagrant
default: SSH auth method: password
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.
```


Attempting to ssh into the machine gives me this error:
```
==> default: The machine you're attempting to SSH into is configured to use
==> default: password-based authentication. Vagrant can't script entering the
==> default: password for you. If you're prompted for a password, please enter
==> default: the same password you have configured in the Vagrantfile.
ssh_exchange_identification: Connection closed by remote host
```

It looks as though 
```
config.ssh.paranoid = false
config.ssh.keys_only = false
``` 
were removed in fd1df3a. I tried manually adding them to my Vagrantfile to see if it made a difference, but it just gave me a new error:

`SSH: * The following settings shouldn't exist: keys_only, paranoid`

**Comment by joshbetz** [comment]
I think this is related to #502

**Comment by Faison** [comment]
Hi there,

Just wanted to add some more details to the mix. I'm running into the same issues described above, and #502 didn't resolve the issue (as I setup this VM after that fix was pushed).

After `vagrant up` hangs while trying to ssh, I `Ctrl+c` the process, then I try to ssh into the VM with `ssh -vvv vagrant@127.0.0.1 -p 2222` and this is what I see:

```
OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014
debug1: Reading configuration data /home/<redacted>/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug2: ssh_connect: needpriv 0
debug1: Connecting to 127.0.0.1 [127.0.0.1] port 2222.
debug1: Connection established.
debug3: Incorrect RSA1 identifier
debug3: Could not load "/home/<redacted>/.ssh/id_rsa" as a RSA1 public key
debug1: identity file /home/<redacted>/.ssh/id_rsa type 1
debug1: identity file /home/<redacted>/.ssh/id_rsa-cert type -1
debug1: identity file /home/<redacted>/.ssh/id_dsa type -1
debug1: identity file /home/<redacted>/.ssh/id_dsa-cert type -1
debug1: identity file /home/<redacted>/.ssh/id_ecdsa type -1
debug1: identity file /home/<redacted>/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/<redacted>/.ssh/id_ed25519 type -1
debug1: identity file /home/<redacted>/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8
ssh_exchange_identification: read: Connection reset by peer
```
The output there is a little confusing, since its trying to use a private key as a public key. But either way, for more details that may or may not help, here is what is found in etc/ssh/ssh_config for the `*` option:
```
Host *
    SendEnv LANG LC_*
    HashKnownHosts yes
    GSSAPIAuthentication yes
    GSSAPIDelegateCredentials no
```
And one important note, that ssh command does succeed in getting me to the username/password entry step.

So yeah, I don't know if this helps at all, but maybe someone who knows more about systems than me can figure something out with that.

**Comment by Faison** [comment]
I found the solution, though it's a pain to implement: http://serverfault.com/a/482020

1 - So basically, you need to boot up the VM with the VirtualBox GUI. This takes a looooong time.
2 - Once the login screen is up, you can login with the username `vagrant` and the password is the same as the username
3 - You need to open up `/etc/default/grub` with a text editor (like nano)
4 - Add the following to the bottom of the file and save the file: `GRUB_RECORDFAIL_TIMEOUT=2`
5 - run `sudo update-grub`
6 - shutdown the vm
7 - run `vagrant up`

So basically, the image used for VIP Quickstart needs to be updated to have that `GRUB_RECORDFAIL_TIMEOUT=2` line in place, otherwise this issue will continue to pop up.

**Comment by joshbetz** [comment]
@Faison I think that will fix it, but I don't think that's the only way. This doesn't happen every time.

**Comment by Faison** [comment]
That's a good point. I've been dealing with these issues for months, it was only last week that vagrant up stopped working completely for my existing and even new Quickstart environments.

The other two things I've done to solve the intermittent issues are:
1 - Make sure I have more than 2 GB free RAM when I `vagrant up`
2 - Have the vagrant plugin `vagrant-vbguest` installed

That's all I've got, really. Hopefully it helps some people running into the same issues that I've had.

**Comment by norcross** [comment]
I've applied each fix that's been outlined in here, but still having issues. When running a provision, it hangs on the following: `/Stage[main]//Service[networking]/ensure: ensure changed 'stopped' to 'running'` which doesn't really give me any indication as to what's going on.

**Comment by jjeaton** [comment]
I reprovisioned yesterday and was able to `vagrant ssh` by entering the password.

After reloading, now when trying to `vagrant ssh` I get:

```
$ vagrant ssh
==> default: The machine you're attempting to SSH into is configured to use
==> default: password-based authentication. Vagrant can't script entering the
==> default: password for you. If you're prompted for a password, please enter
==> default: the same password you have configured in the Vagrantfile.
ssh_exchange_identification: Connection closed by remote host
```

**Comment by alexcavender** [comment]
@jjeaton Did you ever get this to work? I'm getting the same error

**Comment by jjeaton** [comment]
No, no longer using quickstart since it has been deprecated.

**Comment by tomjn** [comment]
@alexcavender VIP Quickstart is deprecated, we strongly recommend switching to a different local environment using [these instructions](https://vip.wordpress.com/documentation/vip/dev-environment/)
